### PR TITLE
Swap for more-maintained body-scroll-lock package

### PR DESCRIPTION
**Pull request by bhollis**
This could fix the Android body-scroll-lock problem? #5615.

### CODE REVIEW

**Comment by ryan-rushton** [comment]
I will pull it down tonight and give it a test

**Comment by ryan-rushton** [comment]
While the `isAndroid` check seems to be working as expected I think the real issue here is the return of the `useLayoutEffect` that is used for cleanup is getting fired when we don't expect it to. See this

https://github.com/DestinyItemManager/DIM/assets/7344652/e02af45c-246c-4474-9ae3-cfc8363c8f45

**Review by ryan-rushton — APPROVED** [review]
General update and new function looks like it will work fine.

File: src/app/dim-ui/Sheet.tsx [hunk 1]
```diff
@@ -324,7 +324,7 @@ function useLockSheetContents(sheetContents: React.MutableRefObject<HTMLDivEleme
       sheetContents.current = contents;
       if (sheetContents.current) {
         sheetContents.current.addEventListener('touchstart', blockEvents);
-        if (isiOSBrowser()) {
+        if (isiOSBrowser() || isAndroid()) {
           // as-is, body-scroll-lock does not work on on Android #5615
           document.body.classList.add('body-scroll-lock');
           enableBodyScroll(sheetContents.current);
```
Comment (ryan-rushton) [inline src/app/dim-ui/Sheet.tsx:330]: Do we need this?

**Comment by ryan-rushton** [comment]
So I had a quick go at doing exactly that of moving the locking to the sheet. Mostly works but as you can see here the body rendering gets a little weird and something is still overflowing

https://github.com/DestinyItemManager/DIM/assets/7344652/829b4d4c-7b22-4e44-bdef-058c3a6cb67b

**Comment by bhollis** [comment]
I took a stab at slimming this down - let me know if it works better.

**Comment by ryan-rushton** [comment]
Will give it a test tonight and report back

**Comment by bhollis** [comment]
I'll merge and see what happens

**Comment by ryan-rushton** [comment]
Was just sitting down to have a look. I think its pretty safe and I only use emulation for android so sounds good.

**Comment by bhollis** [comment]
I ended up reverting this because it introduced a weird behavior on rounded-corner iPhones - the bottom of the UI jumped up by the amount of the `safe-area-inset-bottom`.

**Pull request merged**